tests/lib/test_archive.py

```diff
@@ -269,7 +273,7 @@ def test_unpack_strip_n_unexpected_structure_inconsistent_components(
         f"{excinfo.value}"
         == """unexpected archive structure:
 
-foo/v2/baz doesn't have the prefix to be removed (foo/v1/)
+trying to strip 2 leading components but foo isn't that deep
```

Comment:
ugh so now that i have a properly formed tarfile it reveals a problem with the implementation

if you try and strip 2, you can't use the first member (`foo/`) to determine stripped_prefix, it'll error like this